src/components/accessories/patientTriage/patientTriageForm/PatientTriageForm.tsx

```diff
@@ -0,0 +1,286 @@
+import React, { FunctionComponent, useCallback, useState } from 'react';
+import { useFormik } from "formik";
+import {
+  formatAllFieldValues,
+  getFromFields,
+} from "../../../../libraries/formDataHandling/functions";
+import DateField from '../../dateField/DateField';
+import { object } from 'yup';
+import { TProps } from './types';
+import ConfirmationDialog from '../../confirmationDialog/ConfirmationDialog';
+import TextButton from '../../textButton/TextButton';
+import SmallButton from '../../smallButton/SmallButton';
+import warningIcon from "../../../../assets/warning-icon.png";
+import TextField from '../../textField/TextField';
+import has from 'lodash.has';
+import get from 'lodash.get';
+import SelectField from '../../selectField/SelectField';
+import "./styles.scss"
+
+const PatientTriageForm: FunctionComponent<TProps> = ({
+  fields,
+  onSubmit,
+  submitButtonLabel,
+  resetButtonLabel,
+  isLoading,
+}) => {
+
+  const validationSchema = object({
+    // TODO  
+  });
+
+  const initialValues = getFromFields(fields, "value");
+  const options = getFromFields(fields, "options");
+
+  const formik = useFormik({
+    initialValues,
+    validationSchema,
+    enableReinitialize: true,
+    onSubmit: (values) => {
+      const formattedValues = formatAllFieldValues(fields, values);
+      onSubmit(formattedValues);
+    },
+  });
+
+  const { setFieldValue, handleBlur } = formik;
+
+  const dateFieldHandleOnChange = useCallback(
+    (value) => {
+      setFieldValue("triageDate", value);
+    },
+    [setFieldValue]
+  );
+
+  const isValid = (fieldName: string): boolean => {
+    return has(formik.touched, fieldName) && has(formik.errors, fieldName);
```

Comment:
This logic seems funny to me 🤔. If the field **is touched AND has an error**, it should **NOT be valid**, right? I guess it should be something like the following:
```suggestion
    return has(formik.touched, fieldName) && !has(formik.errors, fieldName);
```
ps.: If I'm wrong, take it easy. It's Saturday morning haha.

Reply:
I think that it's the variable's name that create some kind of confusion. In fact, isValid is used in order to valorize the props called "error" in order to check if there is an error or not. So I think that here we can just make a name refactoring replacing isValid with hasError or something like that.